src/Aspire.Hosting.Azure/Extensions/AzureAppConfigurationExtensions.cs

```diff
@@ -47,12 +47,9 @@ public static IResourceBuilder<AzureAppConfigurationResource> AddAzureAppConfigu
 
             store.Properties.Tags["aspire-resource-name"] = construct.Resource.Name;
 
-            if (configureResource != null)
-            {
-                var resource = (AzureAppConfigurationResource)construct.Resource;
-                var resourceBuilder = builder.CreateResourceBuilder(resource);
-                configureResource(resourceBuilder, construct, store);
-            }
+            var resource = (AzureAppConfigurationResource)construct.Resource;
+            var resourceBuilder = builder.CreateResourceBuilder(resource);
+            configureResource(resourceBuilder, construct, store);
```

Comment:
We should be throwing an ArgumentNullException if someone passes in null, and not letting a NullReferenceException occur.

Personally, I would just keep the signature `Action<IResourceBuilder<AzureAppConfigurationResource>, ResourceModuleConstruct, AppConfigurationStore>? configureResource`. You can drop the default `= null` if you want.

Then above, you can just pass in `null` instead of a dummy `(_, _, _) => { }`

Reply:
Done. Although I find something deeply unsettling about passing null into a method these days ;)